host.py

```diff
@@ -237,29 +247,30 @@ def ssh_connect(self, username: str, password: Optional[str] = None, rsa_path: s
             pw = PasswordLogin(self._hostname, username, password)
             self._logins.append(pw)
 
+        if discover_auth:
+            auto = AutoLogin(self._hostname, username)
+            self._logins.append(auto)
+
         self.ssh_connect_looped(self._logins)
 
-    def ssh_connect_looped(self, logins: list[Login]) -> None:
-        if len(logins) == 0:
-            raise Exception("No usable logins found")
+    def ssh_connect_looped(self, logins: list[Login], timeout: int = 360) -> None:
+        if not logins:
+            raise RuntimeError("No usable logins found")
 
-        login_details = ", ".join([str(login.vars()) for login in logins])
+        login_details = ", ".join([login.debug_details() for login in logins])
         logger.info(f"Attempting SSH connections on {self._hostname} with logins: {login_details}")
 
-        for _ in range(360):
+        for _ in range(timeout):
             for login in logins:
                 try:
                     self._host = login.login()
                     logger.info(f"Login successful on {self._hostname}")
                     return
-                except ssh_exception.AuthenticationException as e:
-                    logger.error(f"{type(e).__name__} - {str(e)} for login {login.vars()} on host {self._hostname}")
-                    time.sleep(10)
-                except (ssh_exception.NoValidConnectionsError, ssh_exception.SSHException, socket.error, socket.timeout) as e:
-                    logger.error(f"Connection issue for login {login.vars()} on host {self._hostname}: {type(e).__name__} - {str(e)}")
+                except (ssh_exception.AuthenticationException, ssh_exception.NoValidConnectionsError, ssh_exception.SSHException, socket.error, socket.timeout) as e:
+                    logger.error(f"{type(e).__name__} - {str(e)} for login {login.debug_details()} on host {self._hostname}")
```

Comment:
I think this does not qualify an `error`. It hardly qualifies as a `warning`. This can happen regularly, when a machine is rebooting. Warnings/Errors in the log are cause for investigation. There is nothing to investigate, when we are about to retry.

Also, I would remember to only log one message at info level, and subsequent ones at debug level. No need to spam the log. Optimizing the log output seems important to me, as it's the primary mean to understand what happens.

Reply:
That is a good point, I'll add that